pkg/simplegui/simplegui.go

```diff
@@ -0,0 +1,181 @@
+	modal := tview.NewModal()
+	modal.SetTitle(title).
+		SetBorder(true).
+		SetTitleAlign(tview.AlignCenter)
+	modal.SetText(message)
+	if withButton {
+		modal.AddButtons([]string{"OK"}).
+			SetDoneFunc(action)
+	}
+
+	return modal
+}
+
+func BuildTheUi(pl platforms.Platform, running bool, cfg *config.Instance, logDestinationPath string) (*tview.Application, error) {
+	app := tview.NewApplication()
+	modal := tview.NewModal()
+	logExport := tview.NewList()
+
+	var statusText string
+	if running {
+		statusText = "RUNNING"
+	} else {
+		statusText = "NOT RUNNING"
+	}
+
+	ip, err := utils.GetLocalIp()
+	var ipDisplay string
+	if err != nil {
+		ipDisplay = "Unknown"
+	} else {
+		ipDisplay = ip.String()
+	}
+
+	// ugly text for the modal content. sorry.
+	text := ""
+	text = text + "  Visit zaparoo.org for guides and help!  \n"
+	text = text + "──────────────────────────────────────────\n"
+	text = text + "  Service:        " + statusText + "\n"
+	text = text + "  Device address: " + ipDisplay + "\n"
+	text = text + "──────────────────────────────────────────\n"
+
+	pages := tview.NewPages().
+		AddPage("main", modal, true, true)
+
+	// create the small log export modal
+	logExport.
+		AddItem("Upload to termbin.com", "", 'a', func() {
+			pages.RemovePage("export")
+			outcome := uploadLog(pl, pages, app)
+			modal := genericModal(outcome, "Log upload", func(buttonIndex int, buttonLabel string) {
+				pages.RemovePage("upload")
+			}, true)
+			pages.AddPage("upload", modal, true, true)
+		})
+	if logDestinationPath != "" {
+		logExport.AddItem("Copy to SD card", "", 'b', func() {
+			pages.RemovePage("export")
+			outcome := copyLogToSd(pl, logDestinationPath)
+			modal := genericModal(outcome, "Log copy", func(buttonIndex int, buttonLabel string) {
+				pages.RemovePage("copy")
+			}, true)
+			pages.AddPage("copy", modal, true, true)
+		})
+	}
+	logExport.AddItem("Cancel", "", 'q', func() {
+		pages.RemovePage("export")
+	}).
+		ShowSecondaryText(false)
+	// Coloring will require some effort
+	// SetBackgroundColor(modal.GetBackgroundColor())
+	logExport.
+		SetBorder(true).
+		SetBorderPadding(1, 1, 1, 1).
+		SetTitle("Log export")
+
+	// create the main modal
+	modal.SetTitle("Zaparoo Core v" + config.AppVersion + " (" + pl.Id() + ")").
+		SetBorder(true).
+		SetTitleAlign(tview.AlignCenter)
+	modal.SetText(text).
+		AddButtons([]string{"Config", "Export log", "Exit"}).
+		SetDoneFunc(func(buttonIndex int, buttonLabel string) {
+			if buttonLabel == "Exit" {
+				app.Stop()
+			}
+			if buttonLabel == "Config" {
+				configui.ConfigUiBuilder(cfg, app, pages, func() {
+					pages.SwitchToPage("main")
+				})
+			}
+			if buttonLabel == "Export log" {
+				widget := modalBuilder(logExport, 42, 8)
+				pages.AddPage("export", widget, true, true)
+			}
+		})
+
+	return app.SetRoot(pages, true).EnableMouse(true), nil
+}
```

Comment:
this is the code moved, but copyLogToSdCard has been made optional because that requires importing the full mister package